src/transaction/TransactionController.ts

```diff
@@ -1278,23 +1316,21 @@ export class TransactionController extends BaseController<
 
         return [meta, true];
       case TransactionStatus.submitted:
+        if (pollingAttempts === maxPollingAttempts) {
+          const error: Error = new Error(
+            'Transaction failed. The transaction was dropped or replaced by a new one',
+          );
+          this.failTransaction(meta, error);
+          return [meta, false];
+        }
+
         const txObj = await query(this.ethQuery, 'getTransactionByHash', [
           transactionHash,
         ]);
 
-        if (!txObj) {
-          const receiptShowsFailedStatus = await this.checkTxReceiptStatusIsFailed(
-            transactionHash,
-          );
-
-          // Case the txObj is evaluated as false, a second check will
-          // determine if the tx failed or it is pending or confirmed
-          if (receiptShowsFailedStatus) {
-            const error: Error = new Error(
-              'Transaction failed. The transaction was dropped or replaced by a new one',
-            );
-            this.failTransaction(meta, error);
-          }
+        if (!txObj && meta.pollingAttempts) {
```

Comment:
Why is `&& meta.pollingAttempts` in this condition? This is causing it to always be false, because `meta.pollingAtempts` starts at `0` (which is falsy).

```suggestion
        if (!txObj) {
```

Reply:
Seems reasonable!